### PR TITLE
add backend skip parameter

### PR DESCRIPTION
**Pull request by kmbarry1**
This is a step towards allowing [klab](https://github.com/dapphub/klab) to depend directly on the official `evm-semantics`, instead of a fork.

### CODE REVIEW

**Comment by kmbarry1** [comment]
cc @ehildenb

**Comment by dwightguth** [comment]
Is KLab using an upstream commit of K or a fork? If it's using an upstream version of K, you can probably bypass this issue by just not building K from source. I'm not particularly opposed to this change though.

**Comment by kmbarry1** [comment]
`klab` inherits `K` via `evm-semantics`, i.e. it doesn't depend directly on `K`. I think that's intentional to ensure compatibility between the two.

File: Makefile [hunk 1]
```diff
@@ -104,6 +104,9 @@ repo-deps: tangle-deps k-deps plugin-deps
 k-deps: $(K_JAR)
 tangle-deps: $(TANGLER)
 plugin-deps: $(PLUGIN_SUBMODULE)/client-c/main.cpp
+# param for skipping certain K backends, e.g.
+# make BACKEND_SKIP="-Dhaskellbackend.skip -Dllvmbackend.skp" deps
+BACKEND_SKIP=
```
Comment (ehildenb) [inline Makefile:109]:
```suggestion

K_MVN_ARGS :=
ifneq ($(SKIP_LLVM),)
    K_MVN_ARGS += -Dllvm.backend.skip
endif
ifneq ($(SKIP_HASKELL),)
    K_MVN_ARGS += -Dhaskell.backend.skip
endif
```

Comment (ehildenb) [reply Makefile:109]: Then invoke with `make deps RELEASE=true SKIP_LLVM=true SKIP_HASKELL=true`

File: Makefile [hunk 2]
```diff
@@ -115,7 +118,7 @@ else
 endif
 
 $(K_JAR):
-	cd $(K_SUBMODULE) && mvn package -DskipTests -U -Dproject.build.type=$(K_BUILD_TYPE)
+	cd $(K_SUBMODULE) && mvn package -DskipTests -U -Dproject.build.type=$(K_BUILD_TYPE) $(BACKEND_SKIP)
```
Comment (ehildenb) [inline Makefile:121]:
```suggestion
	cd $(K_SUBMODULE) && mvn package -DskipTests -U -Dproject.build.type=$(K_BUILD_TYPE) $(K_MVN_ARGS)
```

File: Makefile (hunk 1 above)
Comment (ehildenb) [reply Makefile:109]: Also consider adding documentation of this in the README.

**Comment by kmbarry1** [comment]
Wait, don't merge, I have documentation updates...

**Comment by kmbarry1** [comment]
Okay added the documentation. Checked building locally both using and without the flags, everything seems to work.

**Pull request merged**